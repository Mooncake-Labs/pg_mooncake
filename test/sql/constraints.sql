CREATE TABLE t (a int, b int NOT NULL, c text NOT NULL) USING columnstore;
INSERT INTO t VALUES (1, 2, 'a'), (3, 4, 'b'), (5, 6, 'c');
INSERT INTO t VALUES (7, NULL, 'd');
INSERT INTO t (a, b) VALUES (7, 8);
INSERT INTO t (b, c) VALUES (8, 'd');
UPDATE t SET b = 0 WHERE a = 1;
UPDATE t SET c = NULL WHERE a = 1;
SELECT * FROM t;
DROP TABLE t;
