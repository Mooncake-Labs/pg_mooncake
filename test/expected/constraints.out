CREATE TABLE t (a int, b int NOT NULL, c text NOT NULL) USING columnstore;
INSERT INTO t VALUES (1, 2, 'a'), (3, 4, 'b'), (5, 6, 'c');
INSERT INTO t VALUES (7, NULL, 'd');
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan) Constraint Error: NOT NULL constraint failed: t.b
INSERT INTO t (a, b) VALUES (7, 8);
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan) Constraint Error: NOT NULL constraint failed: t.c
INSERT INTO t (b, c) VALUES (8, 'd');
UPDATE t SET b = 0 WHERE a = 1;
UPDATE t SET c = NULL WHERE a = 1;
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan) Constraint Error: NOT NULL constraint failed: t.c
SELECT * FROM t;
 a | b | c 
---+---+---
   | 8 | d
 1 | 0 | a
 3 | 4 | b
 5 | 6 | c
(4 rows)

DROP TABLE t;
